src/generator.rs

```diff
@@ -91,9 +91,26 @@ impl<'a> Generator<'a> {
         choices: &'a [T],
         err: impl ToString,
     ) -> Mutation<&T> {
+        if choices.is_empty() {
+            return Err(MutationError::Check("Empty choices".to_string())).into();
+        }
+        if choices.len() == 1 {
+            return Ok(&choices[0]).into();
+        }
+        if self.arb.is_empty() {
+            return Err(MutationError::Check("Ran out of entropy".to_string())).into();
+        }
```

Comment:
I think these need to be Exceptions. Check is for problems with the actual data, other errors are for logic errors in writing the fact itself.

```suggestion
        if choices.is_empty() {
            return Err(MutationError::Exception("Empty choices".to_string())).into();
        }
        if choices.len() == 1 {
            return Ok(&choices[0]).into();
        }
        if self.arb.is_empty() {
            return Err(MutationError::Exception("Ran out of entropy".to_string())).into();
        }
```